File: cpp/121-130/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        
        int minimum(INT_MAX),result(0);
        
        for(vector<int>::iterator it = prices.begin(); it != prices.end(); it++)
        {
            minimum = min(minimum, *it);
            result = max(result, *it-minimum);
            
        }
        return result;
    }
};

File: python/121-130/Best Time to Buy and Sell Stock.py
class Solution:
    # @param {integer[]} prices
    # @return {integer}
    def maxProfit(self, prices):
        res = 0
        minimum = 2**31-1
        for e in prices:
            res = max(res, e - minimum)
            minimum = min(minimum, e) 
        return res
